### DISCUSSION TITLE
Using custom fonts in SKTypeface

### DISCUSSION THREAD

**Discussion by pinsonn** [AvaloniaUI/Avalonia#16763] [Q&A]
There are numerous examples of using a .tff font definition file from a 3rd party source like Google Fonts in an Avalonia app. That workflow works great in my app to use the font in text blocks and other UI elements. However, I'd like to use it in an `SKCanvas.DrawText()` call as a `SKTypeface`. Specifying the font family in a call to `SKTypeface.FromFontFamily()` isn't working (it's loading default instead of the specified family), and while the typeface will load by specifying the path directly in a call to `SKTypeface.FromFile()`, the text isn't appearing on the canvas. Anyone have any suggestions?

**Comment by maxkatz6** [comment]
There is no public API to get SKTypeface from the Avalonia types.
But you should be able to use some reflection or UnsafeAccessor (which is safer than a reflection):
```csharp
Typeface avaloniaTypeface = ...
FontManager.Current.TryGetGlyphTypeface(avaloniaTypeface, out var glyphTypeface)
var skTypeface = glyphTypeface._typeface; // access via reflection or UnsafeAccessor
```

**Reply by HandsomeGoldenKnight** [reply]
GlyphTypefaceImpl is a internal class, how could I access _typeface via UnsafeAccessor?
